DQM/TrackingMonitorSource/plugins/AlcaRecoTrackSelector.cc

```diff
@@ -0,0 +1,66 @@
+#include "DataFormats/GsfTrackReco/interface/GsfTrack.h"
+#include "DataFormats/EgammaCandidates/interface/GsfElectron.h"
+#include "DataFormats/EgammaCandidates/interface/Electron.h"
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "DataFormats/TrackReco/interface/HitPattern.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "TLorentzVector.h"
+//#include "RecoEgamma/ElectronIdentification/interface/CutBasedElectronID.h"
+#include "DataFormats/EgammaReco/interface/BasicCluster.h"
+#include "DataFormats/GsfTrackReco/interface/GsfTrack.h"
+
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+#include "TH1.h"
+#include "DQM/TrackingMonitorSource/interface/AlcaRecoTrackSelector.h"
+
+using namespace std;
+using namespace edm;
+
+AlcaRecoTrackSelector::AlcaRecoTrackSelector(const edm::ParameterSet& ps)
+    : parameters_(ps),
+      tracksTag_(parameters_.getUntrackedParameter<edm::InputTag>("trackInputTag", edm::InputTag("generalTracks"))),
+      tracksToken_(consumes<reco::TrackCollection>(tracksTag_)),
+      ptmin_(parameters_.getUntrackedParameter<double>("ptmin", 0.)),
+      pmin_(parameters_.getUntrackedParameter<double>("pmin", 0.)),
+      etamin_(parameters_.getUntrackedParameter<double>("etamin", -4.)),
+      etamax_(parameters_.getUntrackedParameter<double>("etamax", 4.)),
+      nhits_(parameters_.getUntrackedParameter<uint32_t>("nhits", 1)) {
```

Comment:
Why not, much more simply:
```suggestion
    : tracksTag_(ps.getUntrackedParameter<edm::InputTag>("trackInputTag", edm::InputTag("generalTracks"))),
      tracksToken_(consumes<reco::TrackCollection>(tracksTag_)),
      ptmin_(ps.getUntrackedParameter<double>("ptmin", 0.)),
      pmin_(ps.getUntrackedParameter<double>("pmin", 0.)),
      etamin_(ps.getUntrackedParameter<double>("etamin", -4.)),
      etamax_(ps.getUntrackedParameter<double>("etamax", 4.)),
      nhits_(ps.getUntrackedParameter<uint32_t>("nhits", 1)) {
```